lib/network/http_middlewares.go

```diff
@@ -0,0 +1,31 @@
+package network
+
+import (
+	"fmt"
+	"net/http"
+	"runtime/debug"
+
+	"boscoin.io/sebak/lib/network/httputils"
+	"github.com/gorilla/mux"
+)
+
+func RecoverMiddleware(printStack bool) mux.MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			defer func() {
+				if r := recover(); r != nil {
+					err, ok := r.(error)
+					if !ok {
+						err = fmt.Errorf("panic: %v", r)
+					}
+					httputils.WriteJSONError(w, err)
+					log.Error("recover an panic", "err", err)
```

Comment:
If RecoverMiddleware can have `log` argument, we can use `NodeRunner.Log()` for logging.

Reply:
You means that  `network.Middleware` use  `runner/NodeRunner.log` ?

Reply:
At least, `RecoverMiddleware` is using inside `NodeRunner`, so it will be good.